pghoard/transfer.py

```diff
@@ -324,9 +472,20 @@ def handle_upload(self, site, key, file_to_transfer: UploadEvent):
                 metadata = file_to_transfer.metadata.copy()
                 if file_to_transfer.file_size:
                     metadata["Content-Length"] = str(file_to_transfer.file_size)
+                self.upload_tracker.track_upload_event(
+                    file_key=key,
+                    file_type=file_to_transfer.file_type,
+                    file_size=file_to_transfer.file_size,
+                )
+                upload_progress_fn = partial(self.upload_tracker.increment, file_key=key)
                 storage.store_file_object(
-                    key, f, metadata=metadata, upload_progress_fn=file_to_transfer.incremental_progress_callback
+                    key,
+                    f,
+                    metadata=metadata,
+                    upload_progress_fn=lambda n_bytes: upload_progress_fn(n_bytes=n_bytes),
```

Comment:
**comment** Not sure if there is a better way to do this without wrapping the partial with a lambda function